utils.py

```diff
@@ -104,6 +104,13 @@ def publish(temperatures, battery, client, base_topic, device_name):
 
     client.publish("{0}/{1}/battery".format(base_topic, device_name), battery)
 
+def publishPulse2000(temperatures, battery, main_element, client, base_topic, device_name):
+    for i in range(1, 5):
+        if temperatures[i]:
+            client.publish("{0}/{1}/probe{2}".format(base_topic, device_name, i), temperatures[i])
+
+    client.publish("{0}/{1}/battery".format(base_topic, device_name), battery)
+    client.publish("{0}/{1}/heating_element".format(base_topic, device_name), main_element)
```

Comment:
```suggestion
    if main_element:
        client.publish("{0}/{1}/heating_element".format(base_topic, device_name), main_element)
```
If main_element is optional with default None, you can just wrap the call to publish it in an if.
(in publish() )